modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +843,328 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+                                             v_round(v_absdiff(a3, s3)),
+                                             v_round(v_absdiff(a4, s1)),
+                                             v_round(v_absdiff(a5, s2)),
+                                             v_round(v_absdiff(a6, s3)));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<>
+CV_ALWAYS_INLINE int absdiffc_simd_c3_impl<uchar>(const uchar in[], uchar out[],
+                                                  const v_float32& s1, const v_float32& s2,
+                                                  const v_float32& s3, const int length)
+{
+    constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+    constexpr int num_vectors = 12;
+
+    if (length < 3 * nlanes)
+        return 0;
+
+    int x = 0;
+
+    v_float32 vectors[num_vectors];
+
+    for (;;)
+    {
+        for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+        {
+            for (int i = 0; i < num_vectors; ++i)
+            {
+                vectors[i] = v_load_f32(in + x + i * nlanes / 4);
+            }
+
+            vx_store(&out[x], v_pack_u(v_pack(v_round(v_absdiff(vectors[0], s1)),
+                                              v_round(v_absdiff(vectors[1], s2))),
+                                       v_pack(v_round(v_absdiff(vectors[2], s3)),
+                                              v_round(v_absdiff(vectors[3], s1)))));
+
+            vx_store(&out[x + nlanes], v_pack_u(v_pack(v_round(v_absdiff(vectors[4], s2)),
+                                                       v_round(v_absdiff(vectors[5], s3))),
+                                                v_pack(v_round(v_absdiff(vectors[6], s1)),
+                                                       v_round(v_absdiff(vectors[7], s2)))));
+
+            vx_store(&out[x + 2 * nlanes], v_pack_u(v_pack(v_round(v_absdiff(vectors[8], s3)),
+                                                           v_round(v_absdiff(vectors[9], s1))),
+                                                    v_pack(v_round(v_absdiff(vectors[10], s2)),
+                                                           v_round(v_absdiff(vectors[11], s3)))));
+        }
+
+        if (x < length && (in != out))
+        {
+            x = length - 3 * nlanes;
+            continue;  // process one more time (unaligned tail)
+        }
+        break;
+    }
+    return x;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c3(const T in[], const float scalar[], T out[], int width)
+{
+    constexpr int chan = 3;
+    constexpr int size = static_cast<int>(v_float32::nlanes) + 2;
+    int length = width * chan;
+
+    float init[size];
+    for (int i = 0; i < size; ++i)
+    {
+        init[i] = *(scalar + i % chan);
```

Comment:
@dmatveev AFAIK, Fluid backend performs per-row processing.
So it make sense to implement support for initializer code of such constants.

Reply:
Ok. Scratch buffer was applied.